Feature/seperate scripts

flaviogheri opened the pull request:
Created seperate machine learning scripts and created config file. Finally added the original notebook in repository. 

TODO:
> Create documentation for each of the files
> Check versioning of libraries and make sure those are correct
> Train model
> Check for bugs in later stages after training of model

flaviogheri committed:
created preprocessing script (raw) from kaggle

flaviogheri committed:
created config.yml and added it to preprocess

flaviogheri committed:
created preprocess script

flaviogheri committed:
changed preprocess_data.py into build_features.py and created model_definition script

flaviogheri committed:
finished model definition and model training scripts

flaviogheri committed:
created predict_model.py module

flaviogheri committed:
added original notebook

SDuveen commented on src/models/predict_model.py:
```diff
@@ -0,0 +1,55 @@
+# UNSURE ABOUT THIS SECTION :: DO I NOT WANT TO JUST HAVE THIS OUTPUT WHATEVER
+# INPUT ONE GIVES INSTEAD OF DOING IT FOR THE TEST DATA ONLY ?
+
+import numpy as np
+import seaborn as sns
+import matplotlib.pyplot as plt
+from sklearn.metrics import classification_report, confusion_matrix, accuracy_score
+from tensorflow.keras.models import load_model
+import yaml
```
Maybe group these imports how you did before for clarity

SDuveen commented on src/models/train_model.py:
```diff
@@ -0,0 +1,39 @@
+params = config['params']
+
+
+# UKNOWN HERE : Do I load data using np.load or something else !?
+x_train = np.load("data/processed/x_train.npy")
+y_train = np.load("data/processed/y_train.npy")
+x_val = np.load("data/processed/x_val.npy")
+y_val = np.load("data/processed/y_val.npy")
```
As far as I am aware `np.load` is the correct way of loading in `.npy` files

SDuveen commented on src/models/train_model.py:
```diff
@@ -0,0 +1,39 @@
+                validation_data=(x_val, y_val)
+                )
+
+# saving of model (should this be done differently ?)
+model.save("models/phishing_model.h5")
```
I think this should work for saving the model. Does this also work if the directory doesn't exist?

SDuveen commented on src/models/predict_model.py:
```diff
@@ -0,0 +1,55 @@
+model = load_model("models/phishing_model.h5")
+
+
+### Part below : Believed to be unnecessary ...
```
Are you talking about all code below this part?

SDuveen commented on src/models/predict_model.py:
```diff
@@ -0,0 +1,55 @@
+# models/predict_model.py
+
+
+# UNSURE ABOUT THIS SECTION :: DO I NOT WANT TO JUST HAVE THIS OUTPUT WHATEVER
+# INPUT ONE GIVES INSTEAD OF DOING IT FOR THE TEST DATA ONLY ?
```
It really depends on what the purpose of this file is supposed to be. If it is just to give us statistics for the model, this is okay. If the purpose is to predict on new data, then we indeed should predict on given input in my opinion

flaviogheri committed:
saving figures and results into reports/figures section

fynian19 committed:
Created init file, added dataset functionality and fixed directory issues

Created init file that runs all files sequentially. added datasets to the files. We might want to update this to a cloud based dataset functionality. and fixed issues with directories in the loading of the dataset and processing in build_features.py

flaviogheri committed:
changes raw file names reference

flaviogheri committed:
fixed config path and changed path for build_features

flaviogheri committed:
fixed typo in train_model

flaviogheri committed:
fixed bugs up to train_model.py

fynian19 committed:
Fixed config file functionality in model_definition and directory fixes train_model

fynian19 committed:
Train model working and predict model working. Directories fixed and __init file runable. first results added for quick reference

fynian19 committed:
Update README.md with current run instructions and python script descriptions
